renderdoc/driver/vulkan/vk_overlay.cpp

```diff
@@ -308,6 +308,8 @@ void VulkanDebugManager::PatchFixedColShader(VkShaderModule &mod, float col[4])
         alias.data[it + 3] = col[3];
       else
         RDCERR("Unexpected constant value");
+
+      patched++;
```

Comment:
Good catch here, I completely messed up on this check (not sure how I missed that). The increment of `patched` though should happen inside the if statements where the constants are actually patched. The intention was to catch the case where somehow less than 4 constants were found so the patching didn't succeed but there wasn't an "Unexpected constant value" error.

Reply:
I figured since the else block was an error, and all the if blocks actually did patching, it made more sense to put it in one place and check that at least the outer if inside the loop caught 4 cases.
P.S. I had no clue what this is actually doing, I just took a random stab at how it was meant to look.

Reply:
Yeh actually that's fair enough, either way there is an error indicating that something went wrong.